lib/exercises.rb

```diff
@@ -23,8 +85,71 @@ def top_k_frequent_elements(list, k)
 #   Each element can either be a ".", or a digit 1-9
 #   The same digit cannot appear twice or more in the same 
 #   row, column or 3x3 subgrid
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) because at method must go through the table 3 times to check for each sceniario to be valid (row, column, 3x3 grid)
+# Space Complexity: O(1) because the size of the variables created is constant. (if the grid is greater than 9x9 this might change for the corners array)
 def valid_sudoku(table)
```

Comment:
This works, but the method is pretty long and could use some refactoring.   Maybe breaking some things into helper methods.